### ISSUE TITLE
SimPagedMemory throwing AttributeError in concrete_parts() and dbg_print()

### ISSUE THREAD

**Issue by RayFoulk** [angr/simuvex#153]
File "simuvex/plugins/symbolic_memory.py", line 941, in concrete_parts
    for k,v in self.mem.iteritems():
AttributeError: 'SimPagedMemory' object has no attribute 'iteritems'

  File "simuvex/plugins/symbolic_memory.py", line 953, in dbg_print
    for i, addr in enumerate(self.mem.iterkeys()):
AttributeError: 'SimPagedMemory' object has no attribute 'iterkeys'

Looks like self.mem is assumed to be a dictionary at these points but it (at least in my case) doesn't appear to be as it's created on lines 47-51 as self.mem = SimPagedMemory(

**Comment by zardus** [comment]
Yeah, this changed after the `r/trees` branch, which refactored SimPagedMemory. The new way would be to iterate over all pages (unfortunately, this would be `self.mem._pages.values()`) and then getting the keys for each page (`page.keys()`).
